src/graphql.ts

```diff
@@ -0,0 +1,48 @@
+import * as sourcegraph from 'sourcegraph'
+import { memoizeAsync } from './memoizeAsync'
+
+interface ResolveRepoNameResponse {
+    data?: {
+        repository?: {
+            uri: string
+        }
+    }
+}
+
+/**
+ * Attempts to use the Sourcegraph graphQL API to resolve a repo name
+ * parsed from a resource URI, possibly affected by repositoryPathPattern,
+ * to the repo's URI, unaffected by repositoryPathPattern,
+ * typically of the form "{host}/{nameWithOwner}" (eg. "github.com/sourcegraph/enterprise").
+ *
+ * Example: with `"repositoryPathPattern": "{nameWithOwner}"` set in synced repository
+ * settings for a github.com code host, this would resolve the repo name "sourcegraph/enterprise"
+ * to "github.com/sourcegraph/enterprise".
+ *
+ * If the graphQL request fails (this can happen on private repositories when the Sourcegraph
+ * browser extension points to the public sourcegraph.com instance), the returned Promise
+ * will resolve to the repo name passed as argument.
+ */
+export const resolveRepoName = memoizeAsync(async (name: string): Promise<string> => {
+    try {
+        const result = await sourcegraph.commands.executeCommand<
+            ResolveRepoNameResponse
+        >(
+            'queryGraphQL',
+            `query ResolveRepoURI($name: String!) {
+                repository(name: $name) {
+                    uri
```

Comment:
The `uri` field is deprecated in our API. It's the same as `name`, which makes this query confusing if the name is the input...

I think what we need here is `externalURLs`

Reply:
They're [not the same](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@0b409ee1afafe87cd7b334a3f2043375f112b020/-/blob/cmd/frontend/types/types.go#L36-41):

> Name [...] is the same as URI, unless the user configures a non-default repositoryPathPattern.

We already use `uri` to resolve raw repo names so that we can build definition URLs: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@0b409ee1afafe87cd7b334a3f2043375f112b020/-/blob/shared/src/backend/repo.ts#L10-47

Reply:
Hmm, I was going after the GraphQL documentation, which just says
"DEPRECATED: Use name.".
https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/schema.graphql#L1757-1758

I would assume that the code in shared/ is a legacy usage of a deprecated field, but I wouldn't want to add reliance on a deprecated field to a third party extension. But it is definitely odd that a field is deprecated that is not the same as its replacement. cc @sourcegraph/core-services @sourcegraph/web

Any reason why `externalURLs` is not the right thing here? Since ultimately CodeCov goes by the GitHub repo URL...

Reply:
Related: https://github.com/sourcegraph/sourcegraph/pull/4379